src/vsr/replica.zig

```diff
@@ -1939,6 +2008,203 @@ pub fn ReplicaType(
+            // to avoid unnecessary work in the presence of an asymmetric partition.
+            const requests = std.mem.bytesAsSlice(vsr.BlockRequest, message.body());
+            assert(requests.len > 0);
+
+            request_loop: for (requests) |*request, i| {
+                if (self.grid.faulty(request.block_address, null)) {
+                    log.warn("{}: on_request_blocks: ignoring block request; faulty " ++
+                        "(replica={} address={} checksum={})", .{
+                        self.replica,
+                        message.header.replica,
+                        request.block_address,
+                        request.block_checksum,
+                    });
+                    continue;
+                }
+
+                var reads = self.grid_reads.iterate();
+                while (reads.next()) |read| {
+                    if (read.read.address == request.block_address and
+                        read.read.checksum == request.block_checksum and
+                        read.destination == message.header.replica)
+                    {
+                        log.debug("{}: on_request_blocks: ignoring block request;" ++
+                            " already reading (replica={} address={} checksum={})", .{
+                            self.replica,
+                            message.header.replica,
+                            request.block_address,
+                            request.block_checksum,
+                        });
+                        continue :request_loop;
+                    }
+                }
+
+                const read = self.grid_reads.acquire() orelse {
+                    log.debug("{}: on_request_blocks: ignoring remaining blocks; busy " ++
+                        "(replica={} ignored={}/{})", .{
+                        self.replica,
+                        message.header.replica,
+                        requests.len - i,
+                        requests.len,
+                    });
+                    return;
+                };
+
+                log.debug("{}: on_request_blocks: reading block " ++
+                    "(replica={} address={} checksum={})", .{
+                    self.replica,
+                    message.header.replica,
+                    request.block_address,
+                    request.block_checksum,
+                });
+
+                read.* = .{
+                    .replica = self,
+                    .destination = message.header.replica,
+                    .read = undefined,
+                };
+
+                self.grid.read_block_repair(
+                    on_block_read_repair,
+                    &read.read,
+                    request.block_address,
+                    request.block_checksum,
+                );
+            }
+        }
+
+        fn on_block(self: *Self, message: *const Message) void {
+            maybe(!self.state_machine_opened);
```

Comment:
```suggestion
            maybe(self.state_machine_opened);
```

I feel like semantically maybe already hints at both possibilities.